doc/03_reference/coverage_plan.rst

```diff
@@ -0,0 +1,302 @@
+.. _coverage-plan:
+
+Coverage Plan
+=============
+
+.. note::
+  Work to implement the functional coverage described in this plan is on-going and the plan itself is not yet complete.
+
+.. todo::
+  Branch prediction hasn't yet been considered, this will add more coverage points and alter some others
+
+Introduction
+------------
+Ibex functional coverage is split into two major categories:
+
+* Architectural coverage - which is concerned with instructions being executed and exercising various features of the RISC-V architecture (e.g. PMP) and does not consider the details of how this execution occurs.
+* Microarchitectural coverage - which is concerned with the specifics of the RTL operation, ensuring interesting corner cases are seen along with various micro-architectural events (e.g. the different kinds of stall) and combinations of them.
+
+Architectural coverage is not Ibex specific. It can be determined directly from a trace of executed instructions and is handled by RISC-V DV, details can be found in the `RISC-V DV documentation <https://htmlpreview.github.io/?https://github.com/google/riscv-dv/blob/master/docs/build/singlehtml/index.html#document-coverage_model>`_.
+
+Microarchitectural coverage will probe the Ibex RTL directly and is described here.
+There is some inevitable overlap between architectural and microarchitectural coverage but we aim to minimise it.
+
+Microarchitectural Events and Behaviour
+---------------------------------------
+Below are lists of specific things from the microarchitecture that will be included in functional coverage.
+Each of the points listed below must be covered.
+This will be further combined using cross coverage which is described in the section below.
+
+Instructions
+^^^^^^^^^^^^
+
+Categories
+""""""""""
+Instructions can be grouped into a number of categories.
+Each category represents a different kind of execution.
```

Comment:
nit: would it be better to say something like `each category of instructions exercises a different datapath in the core` ? It's a little confusing upon first read imo